src/braket/aws/aws_device.py

```diff
@@ -271,3 +278,83 @@ def __eq__(self, other):
         return NotImplemented
+
+    @staticmethod
+    def get_devices(
+        arns: Optional[List[str]] = None,
+        names: Optional[List[str]] = None,
+        types: Optional[List[AwsDeviceType]] = None,
+        statuses: Optional[List[str]] = None,
+        provider_names: Optional[List[str]] = None,
+        order_by: str = "name",
+        aws_session: Optional[AwsSession] = None,
+    ) -> List[AwsDevice]:
+        """
+        Get devices based on filters and desired ordering. The result is the AND of
+        all the filters `arns`, `names`, `types`, `statuses`, `provider_names`.
+
+        Examples:
+            >>> AwsDevice.get_devices(provider_names=['Rigetti'], statuses=['ONLINE'])
+            >>> AwsDevice.get_devices(order_by='provider_name')
+            >>> AwsDevice.get_devices(types=['SIMULATOR'])
+
+        Args:
+            arns (List[str], optional): device ARN list, default is `None`
+            names (List[str], optional): device name list, default is `None`
+            types (List[AwsDeviceType], optional): device type list, default is `None`
+            statuses (List[str], optional): device status list, default is `None`
+            provider_names (List[str], optional): provider name list, default is `None`
+            order_by (List[str], optional): field to order result by, default is `name`.
+                Accepted values are ['arn', 'name', 'type', 'provider_name', 'status']
+            aws_session (AwsSession, optional) aws_session: An AWS session object. Default = None.
+
+        Returns:
+            List[AwsDevice]: list of AWS devices
+        """
+        order_by_list = ["arn", "name", "type", "provider_name", "status"]
+        if order_by not in order_by_list:
+            raise ValueError(f"order_by '{order_by}' must be in {order_by_list}")
+        device_regions_set = AwsDevice._get_devices_regions_set(arns, provider_names, types)
+        results = []
+        for region in device_regions_set:
+            aws_session = AwsDevice._copy_aws_session(aws_session, region)
+            results.extend(
+                aws_session.search_devices(
+                    arns=arns,
+                    names=names,
+                    types=types,
+                    statuses=statuses,
+                    provider_names=provider_names,
+                )
+            )
+        arns = set([result["deviceArn"] for result in results])
+        devices = [AwsDevice(arn, aws_session) for arn in arns]
+        devices.sort(key=lambda x: getattr(x, order_by))
+        return devices
+
+    @staticmethod
+    def _get_devices_regions_set(
+        arns: Optional[List[str]],
+        provider_names: Optional[List[str]],
+        types: Optional[List[AwsDeviceType]],
+    ) -> Set[str]:
+        """Get the set of regions to call `SearchDevices` API given filters"""
+        device_regions_set = set(
+            AwsDevice.DEVICE_REGIONS[key][0] for key in AwsDevice.DEVICE_REGIONS
```

Comment:
If we're only using the 0th entries, why should be values of `DEVICE_REGIONS` be lists?

Reply:
This is mostly useful if multiple regions are supported for a device (possible in future), then the check in line 262 would not change the region of the aws_session to the first element of a list of the list of supported regions.